### ISSUE TITLE
Maya Instance Browser

### ISSUE THREAD

**Issue by mottosso** [pyblish/pyblish-magenta#45]
We'll need a way to update instances in Maya without fuzz.

It should handle:
- Listing instances in the current scene
- Visualising the current version from which an instance was made
- Listing available versions of an asset per instance
- Switching between versions

Visually it needn't be much more complicated than..

```
Source asset | Instance | Version
 _________________________________
|_________________________________|
|/ben////////| ben01    | v001    |
| chair      |/ben02////| v002    |
| table      |          | v003    |
| sofa       |          | v004    |
| gunA       |          | v005    |
| hand       |          | v006    |
|            |          | v007    |
|            |          | v008    |
|            |          | v009    |
|            |          | v010    |
|            |          |/v011////|
|            |          | v012    |
|____________|__________|_________|
|                         Update  |
|_________________________________|
```

**Comment by BigRoy** [comment]
I put together a quick mockup in PySide, see here:

![reference_manager](https://cloud.githubusercontent.com/assets/2439881/9425518/882048e4-4913-11e5-953b-e5544345b85b.jpg)

Currently it allows you to easily switch any of the currently loaded items to another version and provides the information about the files. (In design, not in actual functionality)

Missing in the mockup is the _update all_ button (which could also be under a right mouse menu), but I assume update all might be done frequently so having it at the bottom could be a good addition.

What do you think?

**Comment by mottosso** [comment]
Looks great,but what do you think about not including the path? I'm thinking it's not relevant to the task and better suited for a tool-tip or right-click menu, such as a "Open in Explorer" item, if it needs to be there at all (probably not).

**Comment by BigRoy** [comment]
> ...what do you think about not including the path? I'm thinking it's not relevant to the task and better suited for a tool-tip or right-click menu, such as a "Open in Explorer" item, if it needs to be there at all (probably not).

Yes, might be too much info as it stands now.

---

Actually I've been looking up some references just now. (Also made some notes over it). This is from the Shotgun/Psyop pipeline video.

![psyop_asset_manager](https://cloud.githubusercontent.com/assets/2439881/9427700/33718664-498b-11e5-86db-db5dbf93794a.jpg)

I think this looks really good.

**Comment by mottosso** [comment]
Yes that looks great, but let's focus on what is absolutely essential right now, which I think is just updating existing instances to their later versions. The UI you've got is a great Beta 0.1, then we could focus on the asset browser, which is more important I think.

**Comment by BigRoy** [comment]
Here's the code for the draft I had so far:
https://github.com/BigRoy/pyblish-magenta/commit/8a14dd713166aee08e81503863b60f548d86c4d4
